BuildingBlocks/src/BuildingBlocks.Infrastructure/EventBus/AzureServiceBus/EventBusAzureServiceBus.cs

```diff
@@ -43,9 +43,14 @@ public EventBusAzureServiceBus(IServiceBusPersisterConnection serviceBusPersiste
     }
 
     public void Dispose()
+    {
+        Task.Run(async () => await DisposeAsync()).GetAwaiter().GetResult();
```

Comment:
Looks strange to me. I would follow that pattern: [https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#implement-both-dispose-and-async-dispose-patterns](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync#implement-both-dispose-and-async-dispose-patterns)

Reply:
Looks really complicated. Especially if we consider that there will probably never be a case where we actually need to dispose, because, as I said in a previous comment, the disposal will only ever happen when shutting down the application.

Reply:
But if you're interested in it, you can give it a shot. 😉 I need to focus on other things now and I don't know when I will be able to revisit it.